metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/security/AuthorizationServerHelper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.token.AuthorizationServerTokenServices;
+import org.springframework.stereotype.Component;
+import org.springframework.test.web.servlet.request.RequestPostProcessor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Component
+@EnableAuthorizationServer
+public class AuthorizationServerHelper extends AuthorizationServerConfigurerAdapter {
+
+    @Autowired
+    private AuthorizationServerTokenServices tokenservice;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    public RequestPostProcessor bearerToken(final String clientid) {
+        return mockRequest -> {
+            OAuth2AccessToken token = createAccessToken(clientid);
+            mockRequest.addHeader("Authorization", "Bearer " + token.getValue());
+            return mockRequest;
+        };
+    }
+
+    private OAuth2AccessToken createAccessToken(final String clientId) {
+        Map<String, Object> clientNameMap = new LinkedHashMap<>();
```

Comment:
THis is manual creation of clients and testing , use clientDetailsService.loadClientByClientId(clientId) to load clients from InmemoryClientDetailsservice

Reply:
@iyangar Is this fixed?

Reply:
@tskir working on the same

Reply:
It doesn't look like this has been fixed because the fields are still manually set.

Reply:
clientDetailsService.loadClientByClientId() gets all information stored in memory. Implementation is changed.
To simulate the clients in memory I have to manually add all fields.